.github/workflows/production-heartbeat.yml

```diff
@@ -11,28 +11,148 @@ on:
         node: ['lts/*']
     runs-on: ${{ matrix.os.vm }}
     timeout-minutes: 60
     steps:
-      # === Setup. We need to get the code, set up nodejs, and create the results directory. ===
-      - uses: actions/checkout@v4
-        with:
-          ref: 'release'
-      - uses: actions/setup-node@v4
-        with:
-          node-version: ${{ matrix.node }}
-      - uses: actions/setup-java@v4
-        with:
-          distribution: 'temurin'
-          java-version: '11'
-      - run: mkdir smoke-test-results
-
-      - name: Say Hello World
-        id: say_hello_world
+      # 1 Install VS Code and Extension on Ubuntu
+      - name: Install VS Code on Ubuntu
+        if: runner.os == 'Linux'
+        run: |
+          sudo apt update
+          sudo apt install wget gpg -y
+          wget -qO- https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > packages.microsoft.gpg
+          sudo install -o root -g root -m 644 packages.microsoft.gpg /usr/share/keyrings/
+          sudo sh -c 'echo "deb [arch=amd64] https://packages.microsoft.com/repos/vscode stable main" > /etc/apt/sources.list.d/vscode.list'
+          sudo apt update
+          sudo apt install code -y
+
+      - name: Install Salesforce Code Analyzer Extension on Ubuntu
+        if: runner.os == 'Linux'
+        run: |
+          code --install-extension salesforce.sfdx-code-analyzer-vscode
+
+      - name: Verify Extension Installation on Ubuntu
+        if: runner.os == 'Linux'
+        run: |
+          if code --list-extensions | grep -q 'salesforce.sfdx-code-analyzer-vscode'; then
+            echo "Extension installed successfully"
+          else
+            echo "Extension installation failed" && exit 1
+          fi
+
+      # 2 Install VS Code and Extension on Windows
+      # We use chocolatey to install vscode since it gives a reliable path for the location of code.exe
+      # We have also seen Windows to be flaky, so adding addition echo statements.
+      - name: Install VS Code on Windows
+        if: runner.os == 'Windows'
+        run: |
+          Write-Host "Installing Chocolatey..."
+          Set-ExecutionPolicy Bypass -Scope Process -Force;
+          [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072;
+          iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
+          Write-Host "Installing Visual Studio Code using Chocolatey..."
+          choco install vscode -y
+
+      - name: Install Salesforce Code Analyzer Extension on Windows
+        if: runner.os == 'Windows'
+        run: |
+          echo "Installing Code Analyzer Extension..."
+          "/c/Program Files/Microsoft VS Code/bin/code" --install-extension salesforce.sfdx-code-analyzer-vscode
+          echo "Installing Code Analyzer Complete"
+          
+          echo "Waiting for 10 seconds..."
+          sleep 10
```

Comment:
Why do we sleep here? Is the install really not complete?

Reply:
We are doing this on Windows since I noticed that sometimes list-extensions doesn't show the extension right away.